In [29]:
import pandas as pd
import numpy as np

In [30]:
df = pd.read_csv('telco_customer_churn.csv')

In [31]:
df

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7043 rows x 21 columns]

### Welcome to Day 9 Homework. Please answer the questions below!

1. Buanglah baris dengan `tenure` samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan `tenure` TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

2. Bagi customer yang ***TIDAK MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa ***rata-rata*** dari `tenure` mereka?

3. Bagi customer yang ***MEMILIKI PARTNER*** dan ***TIDAK MEMILIKI DEPENDENTS***, berapa maksimal `MonthlyCharges` yang tercata?

4. Berapa banyak customer yang ***TIDAK MEMILIKI PARTNER*** namun ***MEMILIKI DEPENDENTS***?

5. Berapa rata-rata ***TOTAL CHARGES*** dari customer ***NON-SENIOR CITIZEN*** (Senior Citizen = 0) yang memiliki ***PHONE SERVICE*** dan memiliki ***INTERNET SERVICE FIBER OPTIC***?

#### Silakan menuliskan kode untuk menjawab 5 pertanyaan di atas...

In [32]:
df_filtered = df[df['tenure'] != 0]
df_filtered

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
...          ...     ...            ...     ...        ...     ...   
7038  6840-RESVB    Male              0     Yes        Yes      24   
7039  2234-XADUH  Female              0     Yes        Yes      72   
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
0              No  No phone service             DSL             No  ...   
1             Yes                No             DSL            Yes  ...   
2             Yes                No             DSL            Yes  ...   
3              No  No phone service             DSL            Yes  ...   
4             Yes                No     Fiber optic             No  ...   
...           ...               ...             ...            ...  ...   
7038          Yes               Yes             DSL            Yes  ...   
7039          Yes               Yes     Fiber optic             No  ...   
7040           No  No phone service             DSL            Yes  ...   
7041          Yes               Yes     Fiber optic             No  ...   
7042          Yes                No     Fiber optic            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
0                  No          No          No              No  Month-to-month   
1                 Yes          No          No              No        One year   
2                  No          No          No              No  Month-to-month   
3                 Yes         Yes          No              No        One year   
4                  No          No          No              No  Month-to-month   
...               ...         ...         ...             ...             ...   
7038              Yes         Yes         Yes             Yes        One year   
7039              Yes          No         Yes             Yes        One year   
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
0                 Yes           Electronic check          29.85         29.85   
1                  No               Mailed check          56.95        1889.5   
2                 Yes               Mailed check          53.85        108.15   
3                  No  Bank transfer (automatic)          42.30       1840.75   
4                 Yes           Electronic check          70.70        151.65   
...               ...                        ...            ...           ...   
7038              Yes               Mailed check          84.80        1990.5   
7039              Yes    Credit card (automatic)         103.20        7362.9   
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
0       No  
1       No  
2      Yes  
3       No  
4      Yes  
...    ...  
7038    No  
7039    No  
7040    No  
7041   Yes  
7042    No  

[7032 rows x 21 columns]

In [33]:
df['Partner'].isin(['No']) & df['Dependents'].isin(['No'])

0       False
1        True
2        True
3        True
4        True
        ...  
7038    False
7039    False
7040    False
7041    False
7042     True
Length: 7043, dtype: bool

In [34]:
df[ (df['Partner'].isin(['No'])) & (df['Dependents'].isin(['No']))]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
1     5575-GNVDE    Male              0      No         No      34   
2     3668-QPYBK    Male              0      No         No       2   
3     7795-CFOCW    Male              0      No         No      45   
4     9237-HQITU  Female              0      No         No       2   
5     9305-CDSKC  Female              0      No         No       8   
...          ...     ...            ...     ...        ...     ...   
7034  0639-TSIQW  Female              0      No         No      67   
7035  8456-QDAVC    Male              0      No         No      19   
7036  7750-EYXWZ  Female              0      No         No      12   
7037  2569-WGERO  Female              0      No         No      72   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
1             Yes                No             DSL                  Yes  ...   
2             Yes                No             DSL                  Yes  ...   
3              No  No phone service             DSL                  Yes  ...   
4             Yes                No     Fiber optic                   No  ...   
5             Yes               Yes     Fiber optic                   No  ...   
...           ...               ...             ...                  ...  ...   
7034          Yes               Yes     Fiber optic                  Yes  ...   
7035          Yes                No     Fiber optic                   No  ...   
7036           No  No phone service             DSL                   No  ...   
7037          Yes                No              No  No internet service  ...   
7042          Yes                No     Fiber optic                  Yes  ...   

         DeviceProtection          TechSupport          StreamingTV  \
1                     Yes                   No                   No   
2                      No                   No                   No   
3                     Yes                  Yes                   No   
4                      No                   No                   No   
5                     Yes                   No                  Yes   
...                   ...                  ...                  ...   
7034                  Yes                   No                  Yes   
7035                   No                   No                  Yes   
7036                  Yes                  Yes                  Yes   
7037  No internet service  No internet service  No internet service   
7042                  Yes                  Yes                  Yes   

          StreamingMovies        Contract PaperlessBilling  \
1                      No        One year               No   
2                      No  Month-to-month              Yes   
3                      No        One year               No   
4                      No  Month-to-month              Yes   
5                     Yes  Month-to-month              Yes   
...                   ...             ...              ...   
7034                   No  Month-to-month              Yes   
7035                   No  Month-to-month              Yes   
7036                  Yes        One year               No   
7037  No internet service        Two year              Yes   
7042                  Yes        Two year              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
1                  Mailed check          56.95        1889.5    No  
2                  Mailed check          53.85        108.15   Yes  
3     Bank transfer (automatic)          42.30       1840.75    No  
4              Electronic check          70.70        151.65   Yes  
5              Electronic check          99.65         820.5   Yes  
...                         ...            ...           ...   ...  
7034    Credit card (automatic)         102.95       6886.25   Yes  
7035  Bank transfer (automatic)          7

In [35]:
mean_tenure = df['tenure'].mean()
mean_tenure

32.37114865824223

In [36]:
df[ (df['Partner'].isin(['Yes'])) & (df['Dependents'].isin(['No']))]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
0     7590-VHVEG  Female              0     Yes         No       1   
8     7892-POOKP  Female              0     Yes         No      28   
12    8091-TTVAX    Male              0     Yes         No      58   
21    1680-VDCWW    Male              0     Yes         No      12   
23    3638-WEABW  Female              0     Yes         No      58   
...          ...     ...            ...     ...        ...     ...   
7023  1035-IPQPU  Female              1     Yes         No      63   
7024  7398-LXGYX    Male              0     Yes         No      44   
7028  9281-CEDRU  Female              0     Yes         No      68   
7031  3605-JISKB    Male              1     Yes         No      55   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService     MultipleLines InternetService       OnlineSecurity  ...  \
0              No  No phone service             DSL                   No  ...   
8             Yes               Yes     Fiber optic                   No  ...   
12            Yes               Yes     Fiber optic                   No  ...   
21            Yes                No              No  No internet service  ...   
23            Yes               Yes             DSL                   No  ...   
...           ...               ...             ...                  ...  ...   
7023          Yes               Yes     Fiber optic                   No  ...   
7024          Yes               Yes     Fiber optic                  Yes  ...   
7028          Yes                No             DSL                   No  ...   
7031          Yes               Yes             DSL                  Yes  ...   
7041          Yes               Yes     Fiber optic                   No  ...   

         DeviceProtection          TechSupport          StreamingTV  \
0                      No                   No                   No   
8                     Yes                  Yes                  Yes   
12                    Yes                   No                  Yes   
21    No internet service  No internet service  No internet service   
23                     No                  Yes                   No   
...                   ...                  ...                  ...   
7023                  Yes                   No                  Yes   
7024                  Yes                   No                   No   
7028                   No                  Yes                  Yes   
7031                   No                   No                   No   
7041                   No                   No                   No   

          StreamingMovies        Contract PaperlessBilling  \
0                      No  Month-to-month              Yes   
8                     Yes  Month-to-month              Yes   
12                    Yes        One year               No   
21    No internet service        One year               No   
23                     No        Two year              Yes   
...                   ...             ...              ...   
7023                  Yes  Month-to-month              Yes   
7024                   No  Month-to-month              Yes   
7028                   No        Two year               No   
7031                   No        One year               No   
7041                   No  Month-to-month              Yes   

                  PaymentMethod MonthlyCharges  TotalCharges Churn  
0              Electronic check          29.85         29.85    No  
8              Electronic check         104.80       3046.05   Yes  
12      Credit card (automatic)         100.35        5681.1    No  
21    Bank transfer (automatic)          19.80        202.25    No  
23      Credit card (automatic)          59.90        3505.1    No  
...                         ...            ...           ...   ...  
7023           Electronic check         103.50        6479.4    No  
7024    Credit card (automatic)          8

In [37]:
max_monthlycharges = df['MonthlyCharges'].max()
max_monthlycharges

118.75

In [38]:
df[ (df['Partner'].isin(['No'])) & (df['Dependents'].isin(['Yes']))]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
6     1452-KIOVK    Male              0      No        Yes      22   
9     6388-TABGU    Male              0      No        Yes      62   
17    9959-WOFKT    Male              0      No        Yes      71   
29    8773-HHUOZ  Female              0      No        Yes      17   
74    5630-AHZIL  Female              0      No        Yes       3   
...          ...     ...            ...     ...        ...     ...   
6946  9013-AQORL  Female              0      No        Yes      48   
6966  6598-KELSS    Male              0      No        Yes      50   
6979  5351-QESIO    Male              0      No        Yes       1   
6984  6447-GORXK    Male              0      No        Yes      31   
6995  7164-BPTUT    Male              0      No        Yes      57   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...  \
6             Yes               Yes     Fiber optic             No  ...   
9             Yes                No             DSL            Yes  ...   
17            Yes               Yes     Fiber optic            Yes  ...   
29            Yes                No             DSL             No  ...   
74            Yes                No             DSL            Yes  ...   
...           ...               ...             ...            ...  ...   
6946           No  No phone service             DSL             No  ...   
6966           No  No phone service             DSL            Yes  ...   
6979           No  No phone service             DSL             No  ...   
6984           No  No phone service             DSL             No  ...   
6995          Yes               Yes             DSL            Yes  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
6                  No          No         Yes              No  Month-to-month   
9                  No          No          No              No        One year   
17                Yes          No         Yes             Yes        Two year   
29                 No          No         Yes             Yes  Month-to-month   
74                 No         Yes          No             Yes  Month-to-month   
...               ...         ...         ...             ...             ...   
6946              Yes          No          No             Yes  Month-to-month   
6966               No          No         Yes              No        One year   
6979               No          No          No              No  Month-to-month   
6984               No          No          No              No  Month-to-month   
6995              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
6                 Yes    Credit card (automatic)          89.10        1949.4   
9                  No  Bank transfer (automatic)          56.15       3487.95   
17                 No  Bank transfer (automatic)         106.70       7382.25   
29                Yes               Mailed check          64.70        1093.1   
74                Yes  Bank transfer (automatic)          64.50         177.4   
...               ...                        ...            ...           ...   
6946               No    Credit card (automatic)          45.55       2108.35   
6966              Yes  Bank transfer (automatic)          43.05       2208.05   
6979               No               Mailed check          24.20          24.2   
6984               No    Credit card (automatic)          25.00         789.2   
6995               No               Mailed check          89.55       5012.35   

     Churn  
6       No  
9       No  
17      No  
29     Yes  
74      No  
...    ...  
6946    No  
6966    No  
6979    No  
6984    No  
6995    No  

[361 rows x 21 columns]

### Silakan menjawab di bawah ini:

#### Soal 1:
Buanglah baris dengan tenure samadengan 0 (jadi, data yang akan kita analisa di PR ini adalah baris-baris dengan tenure TIDAK samadengan 0). Ada berapa baris yang sekarang tersedia di data kita?

#### Jawaban: 7032

#### Soal 2:
Bagi customer yang TIDAK MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa rata-rata dari tenure mereka?

#### Jawaban: 32.37114865824223

#### Soal 3:
Bagi customer yang MEMILIKI PARTNER dan TIDAK MEMILIKI DEPENDENTS, berapa maksimal MonthlyCharges yang tercata?

#### Jawaban: 118.75

### Soal 4:
Berapa banyak customer yang TIDAK MEMILIKI PARTNER namun MEMILIKI DEPENDENTS?

#### Jawaban: 361 customer

#### Soal 5:
Berapa rata-rata TOTAL CHARGES dari customer NON-SENIOR CITIZEN (Senior Citizen = 0) yang memiliki PHONE SERVICE dan memiliki INTERNET SERVICE FIBER OPTIC?

#### Jawaban: 2283.3004408418656

In [68]:
df[(df['SeniorCitizen'] != 0) & (df['PhoneService'] == 'Yes') & (df['InternetService'] == 'Fiber optic')]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
30    3841-NFECX  Female              1     Yes         No      71   
31    4929-XIHVW    Male              1     Yes         No       2   
50    8012-SOUDQ  Female              1      No         No      43   
53    7495-OOKFY  Female              1     Yes         No       8   
55    1658-BYGOY    Male              1      No         No      18   
...          ...     ...            ...     ...        ...     ...   
7010  0723-DRCLG  Female              1     Yes         No       1   
7015  0666-UXTJO    Male              1     Yes         No      34   
7023  1035-IPQPU  Female              1     Yes         No      63   
7032  6894-LFHLY    Male              1      No         No       1   
7041  8361-LTMKD    Male              1     Yes         No       4   

     PhoneService MultipleLines InternetService OnlineSecurity  ...  \
30            Yes           Yes     Fiber optic            Yes  ...   
31            Yes            No     Fiber optic             No  ...   
50            Yes           Yes     Fiber optic             No  ...   
53            Yes           Yes     Fiber optic             No  ...   
55            Yes           Yes     Fiber optic             No  ...   
...           ...           ...             ...            ...  ...   
7010          Yes           Yes     Fiber optic             No  ...   
7015          Yes            No     Fiber optic             No  ...   
7023          Yes           Yes     Fiber optic             No  ...   
7032          Yes           Yes     Fiber optic             No  ...   
7041          Yes           Yes     Fiber optic             No  ...   

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
30                Yes         Yes          No              No        Two year   
31                Yes          No         Yes             Yes  Month-to-month   
50                 No          No         Yes              No  Month-to-month   
53                 No          No          No              No  Month-to-month   
55                 No          No         Yes             Yes  Month-to-month   
...               ...         ...         ...             ...             ...   
7010               No          No          No              No  Month-to-month   
7015              Yes          No         Yes              No  Month-to-month   
7023              Yes          No         Yes             Yes  Month-to-month   
7032               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   

     PaperlessBilling            PaymentMethod MonthlyCharges  TotalCharges  \
30                Yes  Credit card (automatic)          96.35       6766.95   
31                Yes  Credit card (automatic)          95.50        181.65   
50                Yes         Electronic check          90.25       3838.75   
53                Yes  Credit card (automatic)          80.65         633.3   
55                Yes         Electronic check          95.45       1752.55   
...               ...                      ...            ...           ...   
7010              Yes         Electronic check          74.45         74.45   
7015              Yes  Credit card (automatic)          85.20       2874.45   
7023              Yes         Electronic check         103.50        6479.4   
7032              Yes         Electronic check          75.75         75.75   
7041              Yes             Mailed check          74.40         306.6   

     Churn  
30      No  
31      No  
50      No  
53     Yes  
55     Yes  
...    ...  
7010   Yes  
7015    No  
7023    No  
7032   Yes  
7041   Yes  

[831 rows x 21 columns]

In [79]:
df['TotalCharges'] = pd.to_numeric(df['TotalCharges'], errors='coerce')

In [80]:
mean_totalcharges = df['TotalCharges'].mean()
mean_totalcharges

2283.3004408418656